### PR TITLE
Fix `_validation` output group merging

### PR DESCRIPTION
**Pull request by fmeum**
While https://github.com/bazelbuild/bazel/commit/cd725834eba3edb53b7a7dd9867363ef1fe00e57 made it so that a rule and an aspect can both specify validation outputs, it accidentally didn't actually merge the `_validation` output groups. 

By adding the missing merge logic and having the `ValidateTarget` aspect depend on all other aspects, both a rule and an aspect returning a validation output is now fully supported with `--experimental_use_validation_aspect`.

### CODE REVIEW

**Comment by fmeum** [comment]
@lberki Could you review this as you also reviewed the PR this is based on (https://github.com/bazelbuild/bazel/pull/19630)?

**Comment by fmeum** [comment]
@bazel-io fork 7.4.0

File: src/main/java/com/google/devtools/build/lib/rules/core/ValidateTarget.java [hunk 1]
```diff
@@ -53,6 +53,7 @@ public ConfiguredAspect create(
       RepositoryName toolsRepository)
       throws ActionConflictException, InterruptedException {
     OutputGroupInfo outputGroupInfo = OutputGroupInfo.get(ct);
+    System.err.println(ct + ": " + outputGroupInfo);
```
Comment (lberki) [inline src/main/java/com/google/devtools/build/lib/rules/core/ValidateTarget.java:56]: Please remove this line.

File: src/test/shell/integration/validation_actions_test.sh [hunk 2]
```diff
@@ -553,9 +551,36 @@ EOF
 echo "aspect validation output" > $1
 EOF
   chmod +x aspect/aspect_validation_tool
+}
+
+function test_validation_actions_in_rule_and_aspect_no_use_validation_aspect() {
```
Comment (lberki) [inline src/test/shell/integration/validation_actions_test.sh:556]: Mind testing here that both validation actions (in the rule and in the aspect) were executed?

**Comment by lberki** [comment]
Review done!

File: src/test/shell/integration/validation_actions_test.sh (hunk 2 above)
Comment (fmeum) [reply src/test/shell/integration/validation_actions_test.sh:556]: Good point, done!

**Review by lberki — APPROVED** [review]
Thanks, this is now good to go!

**Comment by lberki** [comment]
This broke internal tests at Google and as such, it got rolled back.

There is hopefully just one test, for which I have a fix out. Once that is submitted and @kotlaja confirms that it was the only one, I'll roll it forward.

**Comment by lberki** [comment]
Rollforward is 4607ad439fe8869c8e8951d953e2d3adb613e6d6. Thanks for your patience!

**Comment by fmeum** [comment]
@bazel-io fork 7.4.0

**Comment by fmeum** [comment]
@lberki Looks like the rollforward has been rolled back, but this time the message is just `[]`. Do you have more details on what broke?

**Comment by lberki** [comment]
I'm still investigating. My guess is that some code at Google (not in Bazel) does naughty things, but I don't have evidence for or against.

**Comment by lberki** [comment]
I got it. The culprit is the part that makes the validation aspect depend on all the other aspects: it turns out that at Google, we have a case where we have a top-level aspects that happens to provide the same output group as a simple target. This worked before because the top-level aspects were not merged with their associated targets, but with this change, the validation aspect depends on both and thus it's a "same output group is provided twice" error.

**Comment by fmeum** [comment]
I see, thanks for the explanation. Happy to exclude certain aspects from the logic if needed, but I guess you are in a better position to fix this.

**Comment by lberki** [comment]
Yeah, I'm currently pondering what to do -- the most logical would be to allow merging output groups but I'll need to do some archaeology to figure out why we don't allow it now.

**Comment by lberki** [comment]
I tracked the original source of this functionality to e2033b1d28d3f17c7e307f62b2b13c3eed72a7f6 and there doesn't seem to be any discussion as to what to do about output groups. I suspect the author just went with the safest thing, which is to not allow merging output groups like that.

So I propose allowing merging output groups from aspects, effectively eliminating the "output group X provided twice" error message.

**Comment by fmeum** [comment]
Sounds good to me! Should I make this change in this PR?

**Comment by lberki** [comment]
Don't worry, I'll do it -- it's a trivial change to it's faster than the import dance.

**Comment by fmeum** [comment]
@lberki Just saw https://github.com/bazelbuild/bazel/commit/55eb33456d98492f4b1f42b1b8bc6379cb99fe0c. It looks like this would result in a crash rather than a checked exception if the merged output groups contain depsets of incompatible orders.

**Comment by lberki** [comment]
Hehe. No, although I admit that I only accidentally got it right. The merging logic always creates as stable order nested set:

https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/analysis/OutputGroupInfo.java#L194

and every other kind of nested set is compatible with that:

https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/collect/nestedset/Order.java#L186

**Comment by fmeum** [comment]
I see, thanks for confirming. I always thought that merging a stable set with one of any other order would give you a set of that other order, but it makes way more sense that the other order is disregarded entirely.

I guess you'll try to roll this forward again? Hope it sticks this time.

**Comment by lberki** [comment]
Yep, it's in internal code review; I've been fighting with a cold since list a week so I haven't been able to nag people to do the reviews as vigorously as I usually od.

**Comment by iancha1992** [comment]
The changes in this PR have been included in [Bazel 7.4.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc1`. Thanks!

**Pull request closed without merging**